XamlAnimatedGif/Animator.cs

```diff
@@ -27,10 +27,13 @@ public abstract class Animator : DependencyObject, IDisposable
         private readonly byte[] _previousBackBuffer;
         private readonly byte[] _indexStreamBuffer;
         private readonly TimingManager _timingManager;
-
+        
+        private readonly bool _cacheFrameDataInMemory;
+        private Dictionary<int, byte[][]> _cachedFrameBytes = new Dictionary<int, byte[][]>();
```

Comment:
Now that I think of it... a `byte[][]` is not a very efficient way to represent pixel data. It's usually represented as a single-dimension `byte[]` (which means you need to do some calculations to find the index for a given pixel (something like `y * width + x`)

Reply:
One thing I tried to do was to make my changes minimal and only adding my bits. In RenderFrameAsync indexBuffer is read from indexStream for each line of each frame. Using a byte[][] allows me to just use _cachedFrameBytes[frameIndex][y] to assign a value to indexBuffer otherwise if I use byte[] then I have to somehow slice that to assign value to indexBuffer.